### DISCUSSION TITLE
',' expected.  TS1005  in /view/stylesmap.d.ts(24,68)   incorrect square brackets

### DISCUSSION THREAD

**Discussion by Spiritvoin** [DefinitelyTyped/DefinitelyTyped#55162] [Issues with a @types package] [answered]
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/6953a2c6882f26120f6de9026829dd3afdfb5771/types/ckeditor__ckeditor5-engine/src/view/stylesmap.d.ts#L24
and 
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/6953a2c6882f26120f6de9026829dd3afdfb5771/types/ckeditor__ckeditor5-engine/src/view/stylesmap.d.ts#L36

### incorrect parentheses should be replaced square brackets with curly braces
replase  
`setReducer(name: string, callback: (notation: any) => [property: string, value: string]): void;`
to
`setReducer(name: string, callback: (notation: any) => {property: string, value: string}): void;`

and
replase
`export type PropertyDescriptor = [property: string, value: string];`
to    
 `export type PropertyDescriptor = {property: string, value: string};`

**Comment by fedemp** [comment, marked as answer]
@Spiritvoin Are you sure about your suggestion for `setReducer`? I have noticed that mine were wrong, but also it seems the right return value is `Array<string[]>` given the example in the docs.

```js
// Simplified reducer that always outputs 4 values which are always present:
stylesProcessor.setReducer( 'margin', margin => {
	 return [
	     [ 'margin', `${ margin.top } ${ margin.right } ${ margin.bottom } ${ margin.left }` ]
	 ]
});
```

https://github.com/ckeditor/ckeditor5/blob/0a9e5a9c34a83324fec2242f2d5c86669c08ed02/packages/ckeditor5-engine/src/view/stylesmap.js#L743-L750

I have created a PR at https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59020

----

About `PropertyDescriptor`, there's no error there. Comments in source code reads:

```js
/**
 * A CSS style property descriptor that contains tuplet of two strings:
 *
 * - first string describes property name
 * - second string describes property value
 *
 *		const marginDescriptor = [ 'margin', '2px 3em' ];
 *		const marginTopDescriptor = [ 'margin-top', '2px' ];
 *
 * @typedef {Array.<String, String>} module:engine/view/stylesmap~PropertyDescriptor
 */
```

**Reply by Spiritvoin** [reply]
@fedemp Your fix looks correct.

**Reply by fedemp** [reply]
Thanks a lot for letting me notice the other error. :)
